docs/ff-concepts/file-handling/displaying-media.md

```diff
@@ -0,0 +1,292 @@
+---
+slug: /concepts/file-handling/displaying-media
+title: Displaying Media
+description: Learn how to display media in FlutterFlow.
+tags: [Media Files, Widget]
+sidebar_position: 2
+keywords: [Media Files, Media Management, Assets, Network, Display Media, Media Widgets]
+---
+
+# Displaying Media
+
+Displaying media efficiently is crucial for enhancing user experience in your FlutterFlow app. Whether you're working with images, audio, video, or PDFs, FlutterFlow provides flexible options for integrating and managing media. This guide covers how to set media sources, customize playback settings, and implement best practices like lazy loading, caching, and BlurHash to optimize performance.
+
+To display media on widgets, navigate to the **Properties Panel** and specify the media source under the **[Media] Type** option (e.g., ImageType, AudioType, VideoType). Here are the available options:
+
+### Network
+
+Enter the URL of the media directly into the **Path** input field. This is for media hosted online.
+
+![dm-network-path.avif](imgs/dm-network-path.avif)
+
+If your media is uploaded to Firebase or Supabase, click **Set from Variable** on the **Path** input field, and select **Source** as **Widget State > Uploaded File URL**.
+
+![dm-uploaded-file.avif](imgs/dm-uploaded-file.avif)
+
+For media uploaded via an API, choose **Source** as **Action Outputs > [Action Output Variable Name] (API Response)**. Ensure that the API response contains the URL of the uploaded file. Learn how to extract the URL using [JSON path](../../resources/control-flow/backend-logic/api/rest-api.md#json-path).
+
+![dm-api.avif](imgs/dm-api.avif)
+
+:::info
+
+To handle scenarios where media takes time to load or fails to load, you can set a placeholder. Click **Set from Variable** on the **Path** field and specify a placeholder URL under the **Default Value** property.
+
+:::
+
+### Asset
+
+Use this option to upload media from your computer or select previously uploaded assets.
+
+### Uploaded File
+
+Selecting this option allows for the dynamic handling of media within your app, accommodating media uploaded by users during app usage. For example, if you simply wish to display an image within your app without uploading it anywhere, or if you'd like to preview the image before sending it to your server through an API, you can achieve this by setting the source as **Widget State -> Uploaded Local File**.
+
+![dm-local-upload.avif](imgs/dm-local-upload.avif)
+
+## AudioPlayer
```

Comment:
add a tip[generated code] admonition and mention the underlying package used here.